acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+
+
+def _get_monthmc_yearic(time: "MaskedArray") -> Tuple[np.ndarray, np.ndarray]:
+    """Extracts the monthmc and yearic by parsing the time variable.
+
+    All masked (missing) data points are ignored.
+
+    :param time: Time variable, units are days since 1858-11-17 00:00:00
+    :type time: MaskedArray
+    :return: Arrays for the months and years based on the day of a hurricane
+    :rtype: Tuple[np.ndarray, np.ndarray]
+    """
+    monthmc = np.zeros(time.shape)
+    yearic = np.zeros(time.shape[0])
+
+    # The units attribute of the time variable
+    days_since = datetime(1858, 11, 17, 0, 0, 0)
+
+    for i in range(0, time.shape[0]):
+        for j in range(0, time.shape[1]):
+            if time[i, j] is not np.ma.masked:
+                day_hurr = days_since + timedelta(time[i, j])
+                monthmc[i, j] = day_hurr.month
+                yearic[i] = day_hurr.year
+
+    return monthmc, yearic
+
+
+def _get_mon_wind(
+    vsmc: "MaskedArray",
+    monthmc: np.ndarray,
+    yearic: np.ndarray,
+    num_rows: int,
+) -> Tuple[List[int], List[int]]:
+    """Extracts the months and max wind speeds.
+
+    :param vsmc: Maximum sustained wind speed from official WMO agency.
+    :type vsmc: MaskedArray
+    :param monthmc: Months
+    :type monthmc: np.ndarray
+    :param yearic: Years
+    :type yearic: np.ndarray
+    :param num_rows: Number of rows in the data matrix
+    :type num_rows: int
+    :return: Array of months and max wind speeds
+    :rtype: Tuple[List[int], List[int]]
+    """
+    mon = []
+    wnd = []
+
+    for i in range(0, num_rows):
+        if OBS_START_YR <= yearic[i] <= OBS_END_YR:
+            mon.append(monthmc[i, 0])
+            wnd.append(np.max(vsmc[i, :]))
+
+    return mon, wnd
+
+
+def _calc_mean_ace(vsmc: "MaskedArray", yearic: np.ndarray, num_rows: int) -> float:
+    """Calculates the mean ace using wind speeds.
+
+    :param vsmc: Maximum sustained wind speed from official WMO agency.
+    :type vsmc: MaskedArray
+    :param yearic: Years
+    :type yearic: np.ndarray
+    :param num_rows: Number of rows in the data matrix
+    :type num_rows: int
+    :return: Mean ace
+    :rtype: float
+    """
+    num_years = OBS_YEARS.size
+    ace = np.zeros(num_years)
+
+    for i in range(0, num_years):
+        for j in range(0, num_rows):
+            if yearic[j] == OBS_YEARS[i]:
+                wind = vsmc[j, :]
+                wind_ts = wind[wind >= 35]
+                ace[i] = ace[i] + np.sum(wind_ts ** 2) / 1e4
+
+    return np.mean(ace)
+
+
+def _calc_ts_intensity_dist(wind_speeds: List[int]) -> np.ndarray:
+    """Calculate TC intensity distribution based on wind speed.
+
+    :param wind_speeds: Wind speeds
+    :type wind_speeds: List[int]
+    :return: Number of storms in each hurricane category (tc intensity distribution)
+    :rtype: np.ndarray
+    """
+    num_categories = 6
+    dist = np.zeros(num_categories)
+
+    for speed in wind_speeds:
+        if speed <= 34:
```

Comment:
Good to have a base case so that the rest of the checks are skipped immediately, rather than going through each if statement if `speed <=34`